### PR TITLE
Initial implementation of passing external datastore parameters

### PR DESCRIPTION
**Pull request by berkayoz**
Added ability to look for certificates for the datastore under the names `$cluster_name-etcd` `$cluster_name-apiserver-etcd-client`
Extended ControlPlane configuration to get `datastoreType` and a secret reference for `datastoreServers`

TODO: The naming should be adjusted to reflect `external-datastore` instead of `etcd` in the code and secret names.

### CODE REVIEW

**Comment by bschimke95** [comment]
please resolve conflict

File: bootstrap/api/v1beta2/ck8sconfig_types.go [hunk 1]
```diff
@@ -60,10 +60,13 @@ type CK8sConfigSpec struct {
 	InitConfig CK8sInitConfiguration `json:"initConfig,omitempty"`
 }
 
-// TODO
-// Will need extend this func when implementing other database options.
+// IsEtcdManaged returns true if the control plane is using k8s-dqlite
```
Comment (neoaggelos) [inline bootstrap/api/v1beta2/ck8sconfig_types.go:63]: end line with dot.

File: pkg/ck8s/config_init.go [hunk 2]
```diff
@@ -56,6 +63,19 @@ func GenerateInitControlPlaneConfig(cfg InitControlPlaneConfig) (apiv1.Bootstrap
...
+		datastoreType = "k8s-dqlite"
+	}
+	out.DatastoreType = ptr.To(datastoreType)
+
+	if datastoreType != "k8s-dqlite" {
+		if v := cfg.DatastoreServers; v != "" {
+			out.DatastoreServers = strings.Split(v, ",")
+		}
+	}
+
```
Comment (neoaggelos) [inline pkg/ck8s/config_init.go:78]:
```go
switch cfg.DatastoreType {
case "", "k8s-dqlite":
  // ...
default:
  // ...
}
```

File: pkg/ck8s/config_init.go [hunk 3]
```diff
@@ -88,12 +108,13 @@ func GenerateInitControlPlaneConfig(cfg InitControlPlaneConfig) (apiv1.Bootstrap
...
-	// TODO(neoaggelos): datastore configuration with external etcd (?)
-	k8sDqlitePort := cfg.ControlPlaneConfig.K8sDqlitePort
-	if k8sDqlitePort == 0 {
-		k8sDqlitePort = 2379
+	if datastoreType == "k8s-dqlite" {
+		k8sDqlitePort := cfg.ControlPlaneConfig.K8sDqlitePort
+		if k8sDqlitePort == 0 {
+			k8sDqlitePort = 2379
+		}
+		out.K8sDqlitePort = ptr.To(k8sDqlitePort)
```
Comment (neoaggelos) [inline pkg/ck8s/config_init.go:116]: add this in the switch above

**Review by neoaggelos — COMMENTED** [review]
nice work, some style things to be addressed

File: pkg/ck8s/config_init.go [hunk 4]
```diff
@@ -17,6 +17,8 @@ type InitControlPlaneConfig struct {
 	ControlPlaneConfig    bootstrapv1.CK8sControlPlaneConfig
 	InitConfig            bootstrapv1.CK8sInitConfiguration
 	PopulatedCertificates secret.Certificates
+	DatastoreType         string
+	DatastoreServers      string
```
Comment (neoaggelos) [inline pkg/ck8s/config_init.go:21]: make this a `[]string`

File: pkg/ck8s/config_init.go [hunk 5]
```diff
@@ -56,6 +63,23 @@ func GenerateInitControlPlaneConfig(cfg InitControlPlaneConfig) (apiv1.Bootstrap
...
+		k8sDqlitePort := cfg.ControlPlaneConfig.K8sDqlitePort
+		if k8sDqlitePort == 0 {
+			k8sDqlitePort = 2379
+		}
+		out.K8sDqlitePort = ptr.To(k8sDqlitePort)
+	default:
+		out.DatastoreType = ptr.To("external")
+		if v := cfg.DatastoreServers; v != "" {
+			out.DatastoreServers = strings.Split(v, ",")
+		}
```
Comment (neoaggelos) [inline pkg/ck8s/config_init.go:80]: the caller should do this

**Review by neoaggelos — APPROVED** [review]

**Pull request merged**